core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -629,7 +680,9 @@ abstract class AbstractFetcherThread(name: String,
 
       val initialLag = leaderEndOffset - offsetToFetch
       fetcherLagStats.getAndMaybePut(topicPartition).lag = initialLag
-      PartitionFetchState(offsetToFetch, Some(initialLag), currentLeaderEpoch, state = Fetching)
+      // We don't expect diverging epochs from the next fetch request, so resetting `lastFetchedEpoch`
```

Comment:
Again it seems safe to keep `lastFetchedEpoch` in sync with the local log. If we have done a full truncation above, then `lastFetchedEpoch` will be `None`, but otherwise it seems like we should set it.

Reply:
Updated